swupd/packs.go

```diff
@@ -142,6 +174,27 @@ func WritePack(w io.Writer, fromManifest, toManifest *Manifest, outputDir, chroo
 		fullChrootDir = filepath.Join(chrootDir, fmt.Sprint(toVersion), "full")
 	}
 
+	// Add all deltas that have not failed.
+	hasDelta := make(map[Hashval]bool)
+	for i := range deltas {
+		d := &deltas[i]
+		if d.Error != nil {
+			info.Warnings = append(info.Warnings, err.Error())
+			continue
+		}
+		var fallback bool
+		fallback, err = copyFromDelta(tw, d)
+		if fallback {
```

Comment:
Took me a while to get the logic of this function with `fallback` here and `fallback` below but here using `hasDelta[d.to.Hash]` if `fallback == false`. It would be helpful to add a few comments to the code explaining that if the copy fails you will try another method below (in the next loop).

Reply:
Reorganized the code to show the relationship between fallback and err in this case, added a comment here and in the other place we use fallback logic.